### DISCUSSION TITLE
InsufficientPermissions - token machine to machine

### DISCUSSION THREAD

**Discussion by Paolo-lab** [iTwin/itwinjs-core#6465] [Q&A] [answered]
hello everyone good evening
I'm trying to access a machine to machine authentication token
I have the client-id and the client_secret, my service has the scope of
 itwins:read itwins:modify projects:read projects:modify sensor-data:read sensor-data:modify

https://ims.bentley.com/connect/token

once you get the token:
I would like to use it in this api here, to determine the users of an itwin, but it tells me that I am not authorized

https://api.bentley.com/accesscontrol/itwins/{id}/members/users

How can I solve this problem, am I missing some steps?

I attach an image of the token on jwt:
<img width="418" alt="error token" src="https://github.com/iTwin/itwinjs-core/assets/57446652/9adbd0f6-3a9f-4324-9155-e4883317e781">

**Comment by aruniverse** [comment, marked as answer]
Has your service client been added to the iTwin members?
